src/main.rs

```diff
@@ -1,99 +1,108 @@
-    fn from_lookup_response(response: &datastore1::LookupResponse) -> Game {
-        // TODO - many unwrap without error checking - go add it
-        Game::from_result_map(response.found.as_ref().unwrap().get(0).unwrap().entity.as_ref().unwrap().properties.as_ref().unwrap())
+impl GoogleToken {
+    #[tokio::main]
+    pub(crate) async fn get_token() -> Result<Self, GCPAuthError> {
+        let authentication_manager = gcp_auth::init().await?;
+        let token = authentication_manager
+            .get_token(&[
+                "https://www.googleapis.com/auth/datastore",
+            ])
+            .await?;
+        Ok(Self(token))
     }
+}
 
-    fn lookup(id: &String) -> Option<Game> { // TODO - Create some trait/struct representing a connection, let it contain the hub and project name
-        let client_secret = oauth::service_account_key_from_file(&String::from(TEST_KEY)).unwrap(); // TODO - handle error - not just unwrap
-        let access = yup_oauth2::ServiceAccountAccess::new(
-            client_secret,
-            hyper::Client::with_connector(hyper::net::HttpsConnector::new(hyper_rustls::TlsClient::new()))
-        );
-        let hub = Datastore::new(
-            hyper::Client::with_connector(hyper::net::HttpsConnector::new(hyper_rustls::TlsClient::new())),
-            access
-        );
-
-        let req = LookupRequest {
-            keys: Some(vec!(Key {
-                path: Some(vec!(PathElement {
-                    kind: Some("Game".to_string()), // TODO - game is the name if the entity - add method on DatastoreEntity that can return the name?
-                    id: Some(id.to_string()),
-                    ..Default::default()
-                })),
-                ..Default::default()
-            })),
-            ..Default::default()
-        };
-
-        let result = hub
-            .projects()
-            .lookup(req, TEST_PROJECT_NAME)
-            .doit();
-
-        match result {
-            Err(_e) => { // TODO - do something with this error?
-                None
-            },
-            Ok(res) => {
-                let ret = Game::from_lookup_response(&res.1);
-                Some(ret)
-            }
-        }
+fn test_get_entity() -> Result<Game, String> {
+
+    let gtoken = GoogleToken::get_token()
+        .map_err(|e: gcp_auth::Error| -> String {"Failed to fetch entity".to_string()})?;
+    let client = Client::new(gtoken);
+
+    let projects = client.projects();
+
+    let key = Key{
+        partition_id: None,
+        path: Some(vec![PathElement {
+            id: Some(5632499082330112),
+            kind: Some("Game".to_string()),
+            name: None,
+        }]),
+    };
+    let req = LookupRequest{
+        keys: Some(vec![key]),
+        read_options: None
+    };
+    let resp: LookupResponse = projects.lookup(req, TEST_PROJECT_NAME).execute()
```

Comment:
This lookup + a bunch of cruft above should be simplified by code in `datastore_entity` and just return a single or a list of structs, right?